### PR TITLE
cloud: add developer guides

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Icemap — APPROVED** [review]
LGTM

**Review by Icemap — APPROVED** [review]
LGTM

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 1]
```diff
@@ -154,7 +167,7 @@ mysql --connect-timeout 15 -u root -h <host> -P 4000 -p
   mysqlDataSource.setServerName("localhost");
   mysqlDataSource.setPortNumber(4000);
   mysqlDataSource.setDatabaseName("test");
-  mysqlDataSource.setUser("root");
+  mysqlDataSource.setUser("<prefix>.root");
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:170]: L152-L160 The `Local default cluster` tab is unnecessary.

File: develop/dev-guide-tidb-crud-sql.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ This document briefly introduces how to use TiDB's CURD SQL.
 
 ## Before you start
 
-Please make sure you are connected to a TiDB cluster. If not, refer to [Build a TiDB Cluster in TiDB Cloud (DevTier)](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-free-cluster) to create a free cluster.
+Please make sure you are connected to a TiDB cluster. If not, refer to [Build a TiDB Cluster in TiDB Cloud (Developer Tier)](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-free-cluster) to create a free cluster.
```
Comment (Oreoxmt) [inline develop/dev-guide-tidb-crud-sql.md:12]: - L16-18 is unnecessary
- L38 The document `[Explore SQL with TiDB](/basic-sql-operations.md)` might be not accessible in TiDB Cloud.

Comment (qiancai) [reply develop/dev-guide-tidb-crud-sql.md]: Why it is unnecessary?

File: develop/dev-guide-overview.md [hunk 3]
```diff
@@ -8,7 +8,13 @@ aliases: ['/appdev/dev/app-dev-overview','/tidb/stable/dev-guide-outdated-for-la
 
 This guide is written for application developers, but if you are interested in the inner workings of TiDB or want to get involved in TiDB development, read the [TiDB Kernel Development Guide](https://pingcap.github.io/tidb-dev-guide/) for more information about TiDB.
 
-This tutorial shows how to quickly build an application using TiDB, the possible use cases of TiDB and how to handle common problems. Therefore, before reading this page, it is recommended that you read the [Quick Start Guide for the TiDB Database Platform](/quick-start-with-tidb.md).
+This tutorial shows how to quickly build an application using TiDB, the possible use cases of TiDB and how to handle common problems.
```
[the remark this answers is not in the document]
Comment (qiancai) [reply develop/dev-guide-overview.md:11]: LGTM

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 4]
```diff
@@ -1,36 +1,49 @@
...
 
 7. Click **Connect** to create a traffic filter (a list of client IPs allowed for TiDB connection).
 
     ![connect](/media/develop/IMG_20220331-232726165.png)
 
-8. In the popup window, click **Add Your Current IP Address** to fill in your current IP address, and then click **Create Filter** to create a traffic filter.
-9. Copy the string to connect with a SQL client for later use.
+8. In the popup window, locate **Step 2: Connect with a SQL client**, and then copy the string to connect with a SQL client for later use.
 
     ![SQL string](/media/develop/IMG_20220331-232800929.png)
```
[the remark this answers is not in the document]
Comment (qiancai) [reply develop/dev-guide-build-cluster-in-cloud.md]: Done.

File: develop/dev-guide-build-cluster-in-cloud.md (hunk 1 above)
Comment (qiancai) [reply develop/dev-guide-build-cluster-in-cloud.md]: I've checked and confirmed with Icemap. He will make similar changes in the Chinese docs as well.

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 5]
```diff
@@ -7,29 +7,59 @@ summary: Learn how to build a TiDB cluster in TiDB Cloud (Developer Tier) and co
...
-7. Click **Connect** to create a traffic filter (a list of client IPs allowed for TiDB connection).
+    <CustomContent platform="tidb-cloud">
 
-    ![connect](/media/develop/IMG_20220331-232726165.png)
+    > **Note:**
+    >
+    > For [Developer Tier clusters](/tidb-cloud/select-cluster-tier.md#developer-tier), when you connect to your cluster, you must include the prefix for your cluster in the user name and wrap the name with quotation marks. For more information, see [User name prefix](/tidb-cloud/select-cluster-tier.md#user-name-prefix).
 
-8. In the popup window, click **Add Your Current IP Address** to fill in your current IP address, and then click **Create Filter** to create a traffic filter.
-9. Copy the string to connect with a SQL client for later use.
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:61]:
```suggestion
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 6]
```diff
@@ -7,29 +7,58 @@ summary: Learn how to build a TiDB cluster in TiDB Cloud (Developer Tier) and co
...
+3. To create a Developer Tier cluster for one year free, you can either select the **Developer Tier** plan on the [plan page](https://tidbcloud.com/console/plans) or click **Create Cluster** on the [**Active Clusters**](https://tidbcloud.com/console/clusters) page.
 4. On the **Create Cluster** page, set up your cluster name, cloud provider (for now, only AWS is available for Developer Tier), and region (a nearby region is recommended). Then click **Create** to create your cluster.
-5. Your TiDB Cloud cluster will be created in approximately 5 to 15 minutes. You can check the creation progress at [Active Clusters](https://tidbcloud.com/console/clusters).
-6. After creating a cluster, on the **Active Clusters** page, click the name of your newly created cluster to navigate to the cluster control panel.
 
-    ![active clusters](/media/develop/IMG_20220331-232643794.png)
+    The cluster creation process starts and the **Security Settings** dialog box is displayed.
+
+6. In the **Security Settings** dialog box, set the root password and allowed IP addresses to connect to your cluster, and then click **Apply**.
```
Comment (Icemap) [inline develop/dev-guide-build-cluster-in-cloud.md:33]:
```suggestion
5. In the **Security Settings** dialog box, set the root password and allowed IP addresses to connect to your cluster, and then click **Apply**.
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 7]
```diff
@@ -7,29 +7,58 @@ summary: Learn how to build a TiDB cluster in TiDB Cloud (Developer Tier) and co
...
+6. After creating a cluster, click **Connect** in the upper-right corner. A connection dialog box is displayed.
 
-    ![connect](/media/develop/IMG_20220331-232726165.png)
+    > **Tip:**
+    >
+    > Alternatively, you can also click the name of your newly created cluster to go to the cluster details page, and then click **Connect** in the upper-right corner.
 
-8. In the popup window, click **Add Your Current IP Address** to fill in your current IP address, and then click **Create Filter** to create a traffic filter.
-9. Copy the string to connect with a SQL client for later use.
+8. In the dialog box, locate **Step 2: Connect with a SQL client**, and then copy the string to connect with a SQL client for later use.
```
Comment (Icemap) [inline develop/dev-guide-build-cluster-in-cloud.md:43]:
```suggestion
7. In the dialog box, locate **Step 2: Connect with a SQL client**, and then copy the string to connect with a SQL client for later use.
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 8]
```diff
@@ -136,51 +165,37 @@ mysql --connect-timeout 15 -u root -h <host> -P 4000 -p
...
 
-  </div>
+    {{< copyable "" >}}
 
-  <div label="Non-local default cluster, TiDB Cloud, or other remote cluster">
+    ```java
+    mysqlDataSource.setServerName("localhost");
+    mysqlDataSource.setPortNumber(4000);
+    mysqlDataSource.setDatabaseName("test");
+    mysqlDataSource.setUser("<prefix>.root");
```
Comment (Icemap) [inline develop/dev-guide-build-cluster-in-cloud.md:176]:
```suggestion
    mysqlDataSource.setUser("root");
```

Comment (Icemap) [reply develop/dev-guide-build-cluster-in-cloud.md:176]: It's the origin code.

File: develop/dev-guide-hybrid-oltp-and-olap-queries.md [hunk 9]
```diff
@@ -258,7 +258,18 @@ For more information about how TiDB chooses to use TiFlash, see [Use TiDB to rea
...
+<CustomContent platform="tidb">
+
 - [Quick Start with HTAP](/quick-start-with-htap.md)
 - [Explore HTAP](/explore-htap.md)
+
+</CustomContent>
+
+<CustomContent platform="tidb-cloud">
+
+- [TiDB Cloud HTAP Quick Start](/tidb-cloud/tidb-cloud-htap-quickstart.md)
```
Comment (Icemap) [inline develop/dev-guide-hybrid-oltp-and-olap-queries.md:270]: Will it render the same unsorted list as the above items?

**Review by Icemap — APPROVED** [review]
Rest LGTM

File: develop/dev-guide-hybrid-oltp-and-olap-queries.md (hunk 9 above)
Comment (qiancai) [reply develop/dev-guide-hybrid-oltp-and-olap-queries.md:270]: yes

**Pull request merged**